### PR TITLE
Feat/proposal merkles

### PR DESCRIPTION
**Pull request by gpxl-dev**
Working towards #7

### CODE REVIEW

File: src/features/votes/VoteListItem.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      );
+    });
+  } else {
+    return (
+      <PastEpochCard
+        proposalGroupState={proposalGroupState}
+        proposalGroup={proposalGroup}
+      />
+    );
+  }
```
Comment (piersss) [inline src/features/votes/VoteListItem.tsx:36]:
```suggestion
  }

  return (
    <PastEpochCard
      proposalGroupState={proposalGroupState}
      proposalGroup={proposalGroup}
    />
  );
```

File: src/features/votes/hooks/useGroupMerkleProof.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import { useMemo } from "react";
+import { generateMerkleLeaf } from "../utils/merkleUtils";
+import { useGroupMerkleTree } from "./useGroupMerkleTree";
+
```
Comment (piersss) [inline src/features/votes/hooks/useGroupMerkleProof.ts:4]:
```suggestion

interface MerleTreeProof {
  position: 'left' | 'right';
  data: Buffer;
}

```

File: src/features/votes/hooks/useGroupMerkleProof.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+export const useGroupMerkleProof = (
+  proposalIds: string[],
+  vote: {
+    voter: `0x${string}`;
+    vp: number;
+  },
+) => {
```
Comment (piersss) [inline src/features/votes/hooks/useGroupMerkleProof.ts:11]:
```suggestion
): MerleTreeProof[] => {
```

File: src/features/votes/hooks/useGroupMerkleProof.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  const tree = useGroupMerkleTree(proposalIds);
+
+  return useMemo(() => {
+    if (!tree) return;
```
Comment (piersss) [inline src/features/votes/hooks/useGroupMerkleProof.ts:15]:
```suggestion
    if (!tree) return [];
```

**Review by piersss — CHANGES_REQUESTED** [review]
Looking good. I would define return type of all your components / hooks / utils to make it more clean and easier to maintain.

File: src/features/votes/VoteListItem.tsx (hunk 1 above)
Comment (gpxl-dev) [reply src/features/votes/VoteListItem.tsx:36]: Can you explain why this is better?

I understand the second else is not technically needed, but I personally find it nicer to remind the reader that this is not always what is returned.

File: src/features/votes/hooks/useGroupMerkleProof.ts (hunk 4 above)
Comment (gpxl-dev) [reply src/features/votes/hooks/useGroupMerkleProof.ts:15]: As discussed, returning undefined is intentional

File: src/features/votes/VoteListItem.tsx (hunk 1 above)
Comment (piersss) [reply src/features/votes/VoteListItem.tsx:36]: Because it requires less code and looks better, but that's a personal preference.

**Review by piersss — APPROVED** [review]
LGTM

**Pull request merged**